### DISCUSSION TITLE
How to configure snacks.nvim to open custom shell?

### DISCUSSION THREAD

**Discussion by csoto-3** [LazyVim/LazyVim#5171] [Q&A]
Hey all, I'm just trying to get the popup terminal to open a custom shell (on Win10, open git-bash instead of cmd).
I found the following snippet in the "Config" section of the [snacks.nvim terminal docs](https://github.com/folke/snacks.nvim/blob/main/docs/terminal.md) which should do the trick...

```lua
---@class snacks.terminal.Config
---@field win? snacks.win.Config
---@field override? fun(cmd?: string|string[], opts?: snacks.terminal.Opts) Use this to use a different terminal implementation
{
  win = { style = "terminal" },
}
```

...Where should I put this code snippet? And what part of it needs to be edited to point to the custom shell? And how does it need to be formatted to work with the LazyVim `opts` format?

As a new nvim/lua user (migrating from VSCode), I was able to get the terminal to open on the right side in LazyVim `win = { position = "right" }`; but I've yet to come across something like `---@class` or `---@field` so I'm quite lost.

**Comment by folke** [comment]
check `:h 'shell'`

**Reply by csoto-3** [reply]
Thanks for the tip, I got it working by setting `vim.opt.shell = 'bash'` in my `init.lua` file. And many more thanks for all the work you do developing LazyVim!
